src/middleware/jira-symmetric-jwt-middleware.ts

```diff
@@ -0,0 +1,130 @@
+import { decodeSymmetric, getAlgorithm } from "atlassian-jwt";
+import Logger from "bunyan";
+import { NextFunction, Request, Response } from "express";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
+import { TokenType, verifyQsh } from "~/src/jira/util/jwt";
+import { Installation } from "~/src/models/installation";
+import { moduleUrls } from "~/src/routes/jira/atlassian-connect/jira-atlassian-connect-get";
+import { matchRouteWithPattern } from "~/src/util/match-route-with-pattern";
+
+export const jiraSymmetricJwtMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+
+	if (!await booleanFlag(BooleanFlags.NEW_JWT_VALIDATION, false)) {
+		return next();
+	}
+
+	const token = req.query?.["jwt"] || req.cookies?.["jwt"];
+
+	if (token) {
+		const issuer = getIssuer(token, req.log);
```

Comment:
Should we put a try/catch? If `getIssuer` throw errors, what's the expected output?

Reply:
Refactored the code, it will return 401 if getIssuer `throws` error